### PR TITLE
Transaction panel && E-2396 && B-2710 && E-2372 && B-2711 && E-2373 && B-2712 && E-2380 && B-2713

### PR DESCRIPTION
**Pull request by NeoxAzrot**
b-2712-create-project-transaction-panel

### CODE REVIEW

File: core/domain/reward/reward-contract.types.ts [hunk 1]
```diff
@@ -28,6 +27,10 @@ export type GetRewardsPortParams = HttpClientParameters<{
   QueryParams: GetRewardsQueryParams;
 }>;
 
+/* ----------------------------- Get Rewards CSV ---------------------------- */
+
+export type GetRewardsCsvResponse = HttpStorageResponse<Blob>;
```
Comment (haydencleary) [inline core/domain/reward/reward-contract.types.ts:32]:
```suggestion
export type GetRewardsCsvPortResponse = HttpStorageResponse<Blob>;
```

File: core/domain/project/project-contract.types.ts [hunk 2]
```diff
@@ -94,14 +94,15 @@ export type GetProjectFinancialDetailsByIdPortParams = HttpClientParameters<{
...
   transactions: ProjectTransactionInterface[];
 };
 
-export type GetProjectTransactionsPortResponse = HttpStorageResponse<GetProjectTransactionsResponse>;
-
-type GetProjectTransactionsQueryParams = operations["getProjectTransactions_1"]["parameters"]["query"];
 type GetProjectTransactionsPathParams = operations["getProjectTransactions_1"]["parameters"]["path"];
+type GetProjectTransactionsQueryParams = operations["getProjectTransactions_1"]["parameters"]["query"];
+
+export type GetProjectTransactionsPortResponse = HttpStorageResponse<GetProjectTransactionsResponse>;
```
Comment (haydencleary) [inline core/domain/project/project-contract.types.ts:105]:
```suggestion
export type GetProjectTransactionsPortResponse = HttpStorageResponse<GetProjectTransactionsModel>;
```

File: core/application/react-query-adapter/reward/client/use-get-rewards-csv.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import { RewardFacadePort } from "@/core/domain/reward/input/reward-facade-port";
+
+export function useGetRewardsCsv({
+  pathParams,
+  queryParams,
+  options,
+}: UseMutationFacadeParams<RewardFacadePort["getRewardsCsv"], undefined, Blob>) {
+  const rewardStoragePort = bootstrap.getRewardStoragePortForClient();
+
+  return useMutation(
```
Comment (haydencleary) [inline core/application/react-query-adapter/reward/client/use-get-rewards-csv.ts:17]: If this is a `GET` request why are you using a mutation?

File: app/my-dashboard/_sections/financial-section/_components/transactions-sidepanel/_translations/transaction-sidepanel.en.json [hunk 4]
```diff
@@ -0,0 +1,46 @@
+{
+  "transactions": {
+    "title": "Transactions",
+    "export": "Export CSV",
+    "search": {
+      "placeholder": "Search reward..."
```
Comment (haydencleary) [inline app/my-dashboard/_sections/financial-section/_components/transactions-sidepanel/_translations/transaction-sidepanel.en.json:6]:
```suggestion
      "placeholder": "Search transaction..."
```
This no?

**Review by haydencleary — CHANGES_REQUESTED** [review]

**Review by pixelfact — APPROVED** [review]

**Review by haydencleary — APPROVED** [review]

**Pull request merged**